File: backend/app/schemas/analytics.py
from pydantic import BaseModel


# =========================================================
# ANALYTICS OVERVIEW
# =========================================================

class AnalyticsOverview(BaseModel):
    total_features: int
    enabled_features: int
    disabled_features: int

    total_environments: int
    total_rollouts: int
    total_assignments: int
    active_users: int


class AnalyticsOverviewResponse(BaseModel):
    success: bool
    message: str
    overview: AnalyticsOverview


# =========================================================
# FEATURE ANALYTICS
# =========================================================

class FeatureAnalytics(BaseModel):
    id: int
    name: str
    key: str
    is_enabled: bool

    total_assignments: int
    enabled_assignments: int
    disabled_assignments: int

    total_rollouts: int


class FeatureAnalyticsResponse(BaseModel):
    success: bool
    message: str
    feature: FeatureAnalytics


# =========================================================
# ROLLOUT ANALYTICS
# =========================================================

class RolloutAnalytics(BaseModel):
    id: int
    feature_id: int
    environment_id: int

    percentage: float

    enabled_users: int
    disabled_users: int


class RolloutAnalyticsResponse(BaseModel):
    success: bool
    message: str
    rollout: RolloutAnalytics


# =========================================================
# ASSIGNMENT ANALYTICS
# =========================================================

class AssignmentAnalytics(BaseModel):
    id: int
    user_id: int
    feature_id: int

    is_enabled: bool

    feature_name: str
    user_name: str
    user_email: str


class AssignmentAnalyticsResponse(BaseModel):
    success: bool
    message: str
    assignment: AssignmentAnalytics


# =========================================================
# ENVIRONMENT ANALYTICS
# =========================================================

class EnvironmentAnalytics(BaseModel):
    id: int
    name: str

    total_features: int
    enabled_features: int
    disabled_features: int

    total_rollouts: int


class EnvironmentAnalyticsResponse(BaseModel):
    success: bool
    message: str
    environment: EnvironmentAnalytics

File: frontend/src/types/analytics.ts

// ============================================================
// ANALYTICS TYPES
// ============================================================

// ============================================================
// ANALYTICS OVERVIEW
// ============================================================

export interface AnalyticsOverview {
  total_features: number;

  enabled_features: number;

  disabled_features: number;

  total_environments: number;

  total_rollouts: number;

  total_assignments: number;

  active_users: number;
}

export interface AnalyticsOverviewResponse {
  success?: boolean;

  message?: string;

  overview?: AnalyticsOverview;

  data?: AnalyticsOverview | {
    overview?: AnalyticsOverview;
  };
}

// ============================================================
// FEATURE ANALYTICS
// ============================================================

export interface FeatureAnalytics {
  id: number;

  name: string;

  key: string;

  is_enabled: boolean;

  total_assignments: number;

  enabled_assignments: number;

  disabled_assignments: number;

  total_rollouts: number;
}

export interface FeatureAnalyticsResponse {
  success?: boolean;

  message?: string;

  feature?: FeatureAnalytics;

  data?: FeatureAnalytics | {
    feature?: FeatureAnalytics;
  };
}

// ============================================================
// ROLLOUT ANALYTICS
// ============================================================

export interface RolloutAnalytics {
  id: number;

  feature_id: number;

  environment_id: number;

  percentage: number;

  enabled_users: number;

  disabled_users: number;
}

export interface RolloutAnalyticsResponse {
  success?: boolean;

  message?: string;

  rollout?: RolloutAnalytics;

  data?: RolloutAnalytics | {
    rollout?: RolloutAnalytics;
  };
}

// ============================================================
// ASSIGNMENT ANALYTICS
// ============================================================

export interface AssignmentAnalytics {
  id: number;

  user_id: number;

  feature_id: number;

  environment_id?: number;

  is_enabled: boolean;

  is_active?: boolean;

  value?: string | boolean | number | null;

  feature_name: string;

  user_name: string;

  user_email: string;

  environment_name?: string;
}

export interface AssignmentAnalyticsResponse {
  success?: boolean;

  message?: string;

  assignment?: AssignmentAnalytics;

  data?: AssignmentAnalytics | {
    assignment?: AssignmentAnalytics;
  };
}

// ============================================================
// ENVIRONMENT ANALYTICS
// ============================================================

export interface EnvironmentAnalytics {
  id: number;

  name: string;

  key?: string;

  total_features: number;

  enabled_features: number;

  disabled_features: number;

  total_rollouts: number;
}

export interface EnvironmentAnalyticsResponse {
  success?: boolean;

  message?: string;

  environment?: EnvironmentAnalytics;

  data?: EnvironmentAnalytics | {
    environment?: EnvironmentAnalytics;
  };
}

